src/shadowbox/server/run_action.sh

```diff
@@ -14,20 +14,33 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-echo "** Currently broken, use yarn do shadowbox/docker/run instead **" >&2
-exit 1
-
 do_action shadowbox/server/build
 
+readonly RUN_ID="${RUN_ID:-$(date +%Y-%m-%d-%H%M%S)}"
+readonly RUN_DIR="/tmp/outline/$RUN_ID"
+echo "Using directory $RUN_DIR"
+
 export LOG_LEVEL="${LOG_LEVEL:-debug}"
 export SB_PUBLIC_IP="${SB_PUBLIC_IP:-$(curl https://ipinfo.io/ip)}"
 # WARNING: The SB_API_PREFIX should be kept secret!
 export SB_API_PREFIX=TestApiPrefix
 export SB_METRICS_URL=https://dev.metrics.getoutline.org
-export SB_STATE_DIR=/tmp/outline
+export SB_STATE_DIR="${RUN_DIR}/persisted-state"
+export SB_BIN_DIR="${RUN_DIR}/bin"
+readonly STATE_CONFIG=${SB_STATE_DIR}/shadowbox_server_config.json
+
+[[ -d "${SB_STATE_DIR}" ]] || mkdir -p $SB_STATE_DIR
+[[ -e $STATE_CONFIG ]] || echo '{"hostname":"127.0.0.1"}' > $STATE_CONFIG
+
+# Install third_party dependencies
+OS="linux"
+if [[ `uname` == "Darwin" ]]; then
+  OS="macos"
+fi
+mkdir -p $SB_BIN_DIR
+cp $ROOT_DIR/third_party/prometheus/$OS/prometheus $SB_BIN_DIR/
```

Comment:
Let's make the build command create the entire app directory structure, including these dependencies.

Reply:
Done.